api/app/Http/Controllers/UserController.php

```diff
@@ -138,23 +139,25 @@ public function patchOne($id, Request $request)
         $this->validateId($id);
         $model = $this->repository->find($id);
 
+        // Check if the email is being changed, and initialize confirmation
+        $email = $request->get('email');
+        if ($email && $model->email != $email) {
+            $token = $model->makeConfirmationToken($email, $this->cache);
+            $this->dispatch(new SendEmailConfirmationEmail($model, $email, $token));
+            $request->merge(['email_confirmed' => null]);
+        }
+
         if ($request->get('email_confirmed')) {
             $token = $request->headers->get('email-confirm-token');
             if (!$email = $this->cache->pull('email_confirmation_'.$token)) {
                 throw new ValidationException(
                     new MessageBag(['email_confirmed' => 'The email confirmation token is not valid.'])
                 );
+            } else {
```

Comment:
This should check that the value hasn't changed as the client might send it anyway when the client intends to update another field, and the email token won't be in cache.